dsl/src/main/scala/asobu/dsl/Extractor.scala

```diff
@@ -1,51 +1,112 @@
 package asobu.dsl
 
-import cats.data.{Xor, XorT}
+import cats.data.{Kleisli, Xor, XorT}
 import play.api.mvc.Results._
 import play.api.mvc.{AnyContent, Request, Result}
-import shapeless.ops.hlist.Prepend
-import shapeless.{HList, HNil}
+import shapeless.ops.hlist._
+import shapeless._
 
-import scala.concurrent.Future
+import scala.concurrent.{ExecutionContext, Future}
 import scala.util.{Failure, Success, Try}
-import cats.data.Xor._
+import ExtractResult._
+import cats.syntax.all._
+import CatsInstances._
+import concurrent.ExecutionContext.Implicits.global
+import cats.sequence._
 
-object Extractor {
+object Extractor extends ExtractorFunctions
 
-  def fromTry[Repr <: HList](f: Request[AnyContent] ⇒ Try[Repr]): Extractor[Repr] = { req ⇒
-    Future.successful {
-      f(req) match {
-        case Success(repr) ⇒ Right(repr)
-        case Failure(ex)   ⇒ Left(BadRequest(ex.getMessage))
-      }
-    }
+trait ExtractorOps {
+  import Extractor._
+
+  implicit class extractorOps1[TFrom, Repr <: HList](self: Extractor[TFrom, Repr]) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Class name does not match the regular expression '^[A-Z][A-Za-z0-9]*$'](https://www.codacy.com/app/iHeartRadio/asobu/file/2323061697/issues/source?bid=3037259&fileBranchId=3178877#l22)